### PR TITLE
Support cchardet

### LINKED ISSUE #417: Optionally use cchardet if installed
see https://github.com/PyYoshi/cChardet

### PR DESCRIPTION
**Pull request by asvetlov**
Fix #417

### CODE REVIEW

**Comment by fafhrd91** [comment]
lgtm

**Comment by kxepal** [comment]
+1

File: docs/glossary.rst [hunk 1]
```diff
@@ -13,13 +13,27 @@
...
 
+   chardet
+
+       The Universal Character Encoding Detector
+
+       https://pypi.python.org/pypi/chardet/
+
+   cchardet
+
+       cChardet is high speed universal character encoding detector. -
```
Comment (kxepal) [inline docs/glossary.rst]: oh wait! `encoding detector. -` - there is no need a dot.

Comment (asvetlov) [inline docs/glossary.rst]: Fixed

**Pull request merged**